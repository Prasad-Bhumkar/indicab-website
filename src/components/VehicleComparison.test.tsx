import React from "react";
import { render, screen } from '@testing-library/react';
import _VehicleComparison from './VehicleComparison';
import { describe, expect, test, vitest } from 'vitest';

describe('VehicleComparison Component', (): JSX.Element => {
    const mockVehicles = [
        {
            id: '1',
            name: 'Toyota Camry',
            price: 50
        },
        {
            id: '2',
            name: 'Honda Accord',
            price: 55
        }
    ];

    test('renders comparison with correct vehicle details', (): JSX.Element => {
        render(<_VehicleComparison vehicles={[
            {
                id: '1',
                name: 'Toyota Camry',
                price: 50,
                image: 'toyota-camry.jpg',
                capacity: 4,
                features: ['Air Conditioning', 'Bluetooth']
            },
            {
                id: '2',
                name: 'Honda Accord',
                price: 55,
                image: 'honda-accord.jpg',
                capacity: 4,
                features: ['Air Conditioning', 'Bluetooth']
            }
        ]} />);

        expect(document.body.textContent).toContain('Compare Vehicles');
        expect(screen.getByText('Toyota Camry')).toBeDefined();
        expect(screen.getByText('Honda Accord')).toBeDefined();
        expect(screen.getByText('$50/day')).toBeDefined();
        expect(screen.getByText('$55/day')).toBeDefined();
    });

    test('renders empty state when no vehicles provided', (): JSX.Element => {
        render(<_VehicleComparison vehicles={[]} />);

        expect(screen.getByText('Compare Vehicles')).toBeDefined();
        expect(screen.queryByText('Toyota Camry')).toBeNull();
    });

    test('logs initial distance when provided', (): JSX.Element => {
        const consoleSpy = vitest.spyOn(console, 'log');
        render(<_VehicleComparison vehicles={[
            {
                id: '1',
                name: 'Toyota Camry',
                price: 50,
                image: 'toyota-camry.jpg',
                capacity: 4,
                features: ['Air Conditioning', 'Bluetooth']
            },
            {
                id: '2',
                name: 'Honda Accord',
                price: 55,
                image: 'honda-accord.jpg',
                capacity: 4,
                features: ['Air Conditioning', 'Bluetooth']
            }
        ]} initialDistance={100} />);

        expect(consoleSpy).toHaveBeenCalledWith('Initial distance:', 100);
        consoleSpy.mockRestore();
    });

    test('matches snapshot', (): JSX.Element => {
        const { container } = render(<_VehicleComparison vehicles={[
            {
                id: '1',
                name: 'Toyota Camry',
                price: 50,
                image: 'toyota-camry.jpg',
                capacity: 4,
                features: ['Air Conditioning', 'Bluetooth']
            },
            {
                id: '2',
                name: 'Honda Accord',
                price: 55,
                image: 'honda-accord.jpg',
                capacity: 4,
                features: ['Air Conditioning', 'Bluetooth']
            }
        ]} />);
        expect(container).toMatchSnapshot();
    });
});
