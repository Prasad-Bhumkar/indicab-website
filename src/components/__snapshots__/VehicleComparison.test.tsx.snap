// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VehicleComparison Component > matches snapshot 1`] = `
<div>
  <div
    class="p-6 bg-white rounded-lg shadow"
  >
    <h2
      class="text-2xl font-bold mb-4"
    >
      Compare Vehicles
    </h2>
    <div
      class="grid grid-cols-1 md:grid-cols-3 gap-4"
    >
      <div
        class="border p-4 rounded"
      >
        <h3
          class="font-semibold"
        >
          Toyota Camry
        </h3>
        <p
          class="text-gray-600"
        >
          $
          50
          /day
        </p>
      </div>
      <div
        class="border p-4 rounded"
      >
        <h3
          class="font-semibold"
        >
          Honda Accord
        </h3>
        <p
          class="text-gray-600"
        >
          $
          55
          /day
        </p>
      </div>
    </div>
  </div>
</div>
`;
